mne/preprocessing/ica.py

```diff
@@ -1132,25 +1235,97 @@ def find_bads_ecg(self, inst, ch_name=None, threshold=None, start=None,
             _, p_vals, _ = ctps(sources)
             scores = p_vals.max(-1)
             ecg_idx = np.where(scores >= threshold)[0]
+            # sort indices by scores
+            ecg_idx = ecg_idx[np.abs(scores[ecg_idx]).argsort()[::-1]]
+
+            self.labels_['ecg'] = list(ecg_idx)
+            if ch_name is None:
+                ch_name = 'ECG-MAG'
+            self.labels_['ecg/%s' % ch_name] = list(ecg_idx)
         elif method == 'correlation':
             if threshold is None:
                 threshold = 3.0
-            scores = self.score_sources(
-                inst, target=ecg, score_func='pearsonr', start=start,
-                stop=stop, l_freq=l_freq, h_freq=h_freq,
-                reject_by_annotation=reject_by_annotation, verbose=verbose)
-            ecg_idx = find_outliers(scores, threshold=threshold)
+            self.labels_, scores = self._find_bads_ch(inst, [ecg],
+                                                threshold=threshold,
+                                                start=start, stop=stop,
+                                                l_freq=l_freq, h_freq=h_freq,
+                                                reject_by_annotation=reject_by_annotation, # NOQA
+                                                verbose=verbose,
+                                                prefix="ecg")
         else:
             raise ValueError('Method "%s" not supported.' % method)
-        # sort indices by scores
-        ecg_idx = ecg_idx[np.abs(scores[ecg_idx]).argsort()[::-1]]
-
-        self.labels_['ecg'] = list(ecg_idx)
-        if ch_name is None:
-            ch_name = 'ECG-MAG'
-        self.labels_['ecg/%s' % ch_name] = list(ecg_idx)
         return self.labels_['ecg'], scores
 
+    @verbose
+    def find_bads_ref(self, inst, ch_name=None, threshold=3.0, start=None,
+                      stop=None, l_freq=None, h_freq=None,
+                      reject_by_annotation=True, verbose=None):
+        """Detect MEG reference related components using correlation.
+
+        Detection is based on Pearson correlation between the MEG data
+        components and MEG reference components.
+        Thresholding is based on adaptive z-scoring. The above threshold
+        components will be masked and the z-score will be recomputed
+        until no supra-threshold component remains.
```

Comment:
While we're fixing docstrings, this stuff should really live in `Notes`, too, since it's pretty detailed.